static/app/components/links/link.tsx

```diff
@@ -10,42 +10,48 @@ type AnchorProps = React.HTMLProps<HTMLAnchorElement>;
 type ToLocationFunction = (location: Location) => LocationDescriptor;
 
 type Props = WithRouterProps & {
-  // URL
+  /**
+   * The string path or LocationDescriptor object
+   */
   to: ToLocationFunction | LocationDescriptor;
-  // Styles applied to the component's root
+  /**
+   * Style applied to the component's root
+   */
   className?: string;
 } & Omit<AnchorProps, 'href' | 'target' | 'as' | 'css'>;
 
 /**
  * A context-aware version of Link (from react-router) that falls
  * back to <a> if there is no router present
  */
-class Link extends React.Component<Props> {
-  componentDidMount() {
-    const isRouterPresent = this.props.location;
-    if (!isRouterPresent) {
+const BaseLink: React.FC<Props> = ({location, disabled, to, ref, ...props}) => {
+  useEffect(() => {
+    // check if the router is present
+    if (!location) {
       Sentry.captureException(
         new Error('The link component was rendered without being wrapped by a <Router />')
       );
     }
+  }, [location]);
```

Comment:
I think this is okay since context _can_ go away. Though that also mean this effect will be re-run for every navigation. So I think yeah I will remove it.